src/helpers/url.ts

```diff
@@ -0,0 +1,48 @@
+/**
+ * We currently avoid relying on a large library for URL parsing/building to keep bundle size small,
+ * and avoid env-specific dependencies such as node's `url` or `URL` (which is not available in IE11).
+ *
+ * The helpers here do the bare minimum needed to handle use-cases in this library, and might not
+ * handle more complex scenarios with edge-cases.
+ */
```

Comment:
@OliverJAsh Bear with me 😂 

The reason I went with this route is that I wasn't comfortable shipping a JS SDK that depends on a NodeJS built-in. I didn't want to ship an SDK with a caveat that says "if you're using webpack, you're _most_ likely fine but if you use anything else you'll need a polyfill to use this in the client". I want it to be easily used by everyone, and having to worry about node-js built-ins polyfills could be very daunting for a beginner. 

From there, my options were:
- find an env-agnostic module to do it all
- use `URL` which is supported in both [browsers](https://developer.mozilla.org/en-US/docs/Web/API/URL) and [node](https://nodejs.org/api/url.html)
- write it myself

1- From my searches, I couldn't find an env-agnostic module that was small enough in size.
2- URL seemed promising but it has no IE11 support (not sure how big of an issue that is, but it would be nice to support as much as possible?), but also it meant writing code to identify being in Node VS browser, which meant also choosing whether to handle web workers or not.

I ended up choosing to delegate query logic to a very small lib `querystringify` which handles our needs. It's not comprehensive but is much smaller than alternatives. I then wrote functions to replace `url` and `url-transformers` to cover our use-cases.

I think this approach is safe and a good trade-off. I'm curious what you think though and if you have strong opinions about doing it differently. I'm also open to attempting using `URL`, but once I got it done this way and added tests I figured it's in a good place. WDYT?

Reply:
replying to your other comment here:

I guess I was worried about how safe it is to rely on nodejs built-ins shimming. 

I see your point about us increasing the overall size of users' bundles by recreating code they're most likely already shimming due to all of the other packages they have in their code! Maybe I don't need to worry about any of this and we can just ship it, wait and see if anybody complains!

Reply:
> URL seemed promising

> it meant writing code to identify being in Node VS browser, which meant also choosing whether to handle web workers or not.

Interesting, how come it meant you had to do this? I thought we would just use the global classes in both environments. We wouldn't need any conditional logic, or would we?

I think that would be my favourite approach. Use the modern APIs which are available globally and ask users to polyfill if they need support in old browsers.

Reply:
> We wouldn't need any conditional logic, or would we?

I initially thought we needed to access it on node using `require('url').URL`. Didn't realize it was also available on the global scope! 🤦 

I've reverted my latest changes for now. Will give `url` vs `URL` some more thought

Reply:
Went with `URL`!